glnx-fdio.c

```diff
@@ -205,6 +205,11 @@ glnx_open_tmpfile_linkable_at (int dfd,
     return glnx_throw_errno_prefix (error, "open(O_TMPFILE)");
   if (fd != -1)
     {
+      /* Workaround for https://sourceware.org/bugzilla/show_bug.cgi?id=17523
+       * See also https://github.com/ostreedev/ostree/issues/991
+       */
+      if (fchmod (fd, 0600) < 0)
```

Comment:
It seems like the documentation for this function should be updated, right?

> Note that with O_TMPFILE, the file mode will be `000`; you likely
> want to chmod it before calling glnx_link_tmpfile_at().

Though even before this patch, we do pass in 0600 to `openat`. So maybe you were actually running into that glibc bug at the time as well? With this explitcit `fchmod`, it seems like we can delete that part of the doc.